controllers/restore_controller.go

```diff
@@ -384,8 +384,7 @@ func (r *RestoreReconciler) initVeleroRestores(
 			)
 			return err
 		}
-		// TODO check length of produced name
-		veleroRestore.Name = restore.Name + "-" + veleroBackupName
+		veleroRestore.Name = getValidKsRestoreName(restore.Name, veleroBackupName)
```

Comment:
OK, this is far better than the previous one. Thanks. but we still need to implement the `validationWebHook` to prevent the creation of resources bringing to invalid names.